packages/@lwc/integration-karma/test-hydration/mismatches/class-attr/dynamic-empty-in-ssr-null-string-in-client/index.spec.js

```diff
@@ -0,0 +1,31 @@
+// SSR has no class at all, whereas the client has `class="null"`.
+// This is to test if hydration is smart enough to recognize the difference between a null
+// attribute and the literal string "null".
+export default {
+    props: {
+        className: '',
+    },
+    clientProps: {
+        className: 'null',
+    },
+    snapshot(target) {
+        const p = target.shadowRoot.querySelector('p');
+        return {
+            p,
+            className: p.className,
+        };
+    },
+    test(target, snapshots, consoleCalls) {
+        const p = target.shadowRoot.querySelector('p');
+
+        expect(p).not.toBe(snapshots.p);
+        expect(p.className).not.toBe(snapshots.className);
+
+        expect(consoleCalls.warn).toHaveSize(0);
+        expect(consoleCalls.error).toHaveSize(2);
+        expect(consoleCalls.error[0][0].message).toContain(
+            'Mismatch hydrating element <p>: attribute "class" has different values, expected "null" but found null'
```

Comment:
At first this error message was a bit confusing but I think I got it. It says the client side was expecting the string `'null'` but server side set nothing.

Reply:
Correct. And I think this one would be difficult to clarify further. The server will effectively send `null` in cases of both missing attributes and empty strings, so we don't really know which one the user was using.